app/client/cli/debug.go

```diff
@@ -189,21 +190,24 @@ func initDebug(remoteCLIURL string) {
 			log.Fatalf("[ERROR] Failed to create p2p module: %v", err.Error())
 		}
 		p2pMod = p2pM.(modules.P2PModule)
-		if err != nil {
-			log.Fatalf("[ERROR] Failed to create p2p module: %v", err.Error())
-		}
 
 		// This telemetry module instance is a NOOP because the 'enable_telemetry' flag in the `cfg` above is set to false.
 		// Since this client mimics partial - networking only - functionality of a full node, some of the telemetry-related
 		// code paths are executed. To avoid those messages interfering with the telemetry data collected, a non-nil telemetry
 		// module that NOOPs (per the configs above) is injected.
 		telemetryM, err := telemetry.Create(runtimeMgr)
 		if err != nil {
-			log.Fatalf("[ERROR] Failed to create NOOP telemetry module: " + err.Error())
+			logger.Global.Fatal().Err(err).Msg("Failed to create telemetry module")
 		}
 		telemetryMod := telemetryM.(modules.TelemetryModule)
 
-		_ = shared.CreateBusWithOptionalModules(runtimeMgr, nil, p2pMod, nil, consensusMod, telemetryMod) // TODO: refactor using the `WithXXXModule()` pattern accepting a slice of IntegratableModule
+		loggerM, err := logger.Create(runtimeMgr)
+		if err != nil {
+			logger.Global.Fatal().Err(err).Msg("Failed to create logger module")
+		}
+		loggerMod := loggerM.(modules.LoggerModule)
+
+		_ = shared.CreateBusWithOptionalModules(runtimeMgr, nil, p2pMod, nil, consensusMod, telemetryMod, loggerMod) // TODO: refactor using the `WithXXXModule()` pattern accepting a slice of IntegratableModule
```

Comment:
See https://goldin.io/blog/stop-using-todo ###

```suggestion
		_ = shared.CreateBusWithOptionalModules(runtimeMgr, nil, p2pMod, nil, consensusMod, telemetryMod, loggerMod) // REFACTOR: use the `WithXXXModule()` pattern accepting a slice of IntegratableModule
```